src/Blueprint/Blueprint/Cmdlets/BlueprintArtifacts/NewAzureRMBlueprintArtifact.cs

```diff
@@ -0,0 +1,204 @@
+
+        [Parameter(ParameterSetName = ParameterSetNames.CreateTemplateArtifact, Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = ParameterHelpMessages.ArtifactTemplateParameterFile)]
+        [ValidateNotNullOrEmpty]
+        public string TemplateParameterFile { get; set; }
+
+        [Parameter(ParameterSetName = ParameterSetNames.CreateTemplateArtifact, Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = ParameterHelpMessages.ArtifactTemplateFile)]
+        [ValidateNotNullOrEmpty]
+        public string TemplateFile { get; set; }
+
+        [Parameter(ParameterSetName = ParameterSetNames.CreateRoleAssignmentArtifact, Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = ParameterHelpMessages.ArtifactRoleDefinitionId)]
+        [ValidateNotNullOrEmpty]
+        public string RoleDefinitionId { get; set; }
+
+        [Parameter(ParameterSetName = ParameterSetNames.CreateRoleAssignmentArtifact, Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = ParameterHelpMessages.ArtifactRoleDefinitionPrincipalId)]
+        [ValidateNotNullOrEmpty]
+        public string[] RoleDefinitionPrincipalId { get; set; }
+
+        [Parameter(ParameterSetName = ParameterSetNames.CreatePolicyAssignmentArtifact, Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = ParameterHelpMessages.ArtifactResourceGroup)]
+        [Parameter(ParameterSetName = ParameterSetNames.CreateRoleAssignmentArtifact, Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = ParameterHelpMessages.ArtifactResourceGroup)]
+        [Parameter(ParameterSetName = ParameterSetNames.CreateTemplateArtifact, Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = ParameterHelpMessages.ArtifactResourceGroup)]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        //Alternatively, user can provide a artifactFile
+        [Parameter(ParameterSetName = ParameterSetNames.CreateArtifactByInputFile, Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = ParameterHelpMessages.ArtifactFile)]
+        [ValidateNotNullOrEmpty]
+        public string ArtifactFile { get; set; }
+
+        #endregion
+
+        #region Cmdlet Overrides
+        public override void ExecuteCmdlet()
+        {
+            try
+            {
+                var scope = Blueprint.Scope;
+
+                ThrowIfArtifactExits(scope, Blueprint.Name, Name);
+
+                switch (ParameterSetName)
+                {
+                    case ParameterSetNames.CreateArtifactByInputFile:
+
+                        if (ShouldProcess(Utils.GetDefinitionLocationId(scope), string.Format(Resources.CreateArtifactShouldProcessString, Name)))
+                        {
+                            var artifact = JsonConvert.DeserializeObject<Artifact>(
+                                File.ReadAllText(ResolveUserPath(ArtifactFile)),
```

Comment:
Updated System.IO file and directory references with AzureSession.Instance.DataStore. There is still the Path class I'm using, please let me know if I should replace it too. Otherwise, resolving.